### PR TITLE
Add makegocons -q in the man page

### PR DESCRIPTION
**Pull request by chenglch**
Update the manpage for makegocons.

task: #4848

### CODE REVIEW

File: docs/source/guides/admin-guides/references/man8/makegocons.8.rst [hunk 1]
```diff
@@ -46,17 +44,16 @@ In the case of a hierarchical cluster (i.e. one with service nodes) \ **makegoco
...
 (based on the nodehm.conserver attribute).
 
-For openbmc which uses ssh as the terminal session connection method, goconserver can help save the system
-resources as goconserver could handle the ssh connection within goroutine which is more light-weighted than the command process.
-
-\ **Note:**\  goconserver is an experimental feature, it will not be installed with xcat and will only support the systemd based systems.
-Download and setup the rpm or deb package manually. Release link:
-
+To start goconserver on the specified service node, please setup goconserver package on that service node, then set
+the \ **cosnole**\  column of \ **servicenode**\  table to \ **2**\ .
```
Comment (mattaezell) [inline docs/source/guides/admin-guides/references/man8/makegocons.8.rst]: Typo: console

Comment (chenglch) [reply docs/source/guides/admin-guides/references/man8/makegocons.8.rst]: Done

File: xCAT-client/pods/man8/makegocons.8.pod [hunk 2]
```diff
@@ -44,6 +46,10 @@ Download and setup the rpm or deb package manually. Release link:
 
 Delete rather than add or refresh the nodes specified as a noderange.
 
+=item B<-q|--query>
+
+List the console connection of the nodes. If noderange is not specified, all of the nodes will be displayed.
```
Comment (gurevichmark) [inline xCAT-client/pods/man8/makegocons.8.pod]: Should we say "all of the xCAT defined nodes" ? Or will it be some subset based on node definition attribute ?

File: xCAT-client/pods/man8/makegocons.8.pod [hunk 3]
```diff
@@ -28,13 +26,17 @@ In the case of a hierarchical cluster (i.e. one with service nodes) B<makegocons
...
+
+To support diskless service node, B<makegocons> would help store the console state in the B<consoleenabled> column of B<nodehm>
+table to help recover the console connection when the console storage file is lost after reinstalling the service node.
+
 For openbmc which uses ssh as the terminal session connection method, goconserver can help save the system
 resources as goconserver could handle the ssh connection within goroutine which is more light-weighted than the command process.
 
-B<Note:> goconserver is an experimental feature, it will not be installed with xcat and will only support the systemd based systems.
-Download and setup the rpm or deb package manually. Release link:
+B<Note:> goconserver only support the systemd based systems. It has been integrated with xCAT as a recommend package.
```
Comment (gurevichmark) [inline xCAT-client/pods/man8/makegocons.8.pod]: should be "recommended" instead of "recommend"

File: xCAT-client/pods/man8/makegocons.8.pod [hunk 4]
```diff
@@ -28,13 +26,17 @@ In the case of a hierarchical cluster (i.e. one with service nodes) B<makegocons
 which nodes will have their consoles accessed from the management node and which from a service node
 (based on the nodehm.conserver attribute).
 
+To start goconserver on the specified service node, please setup goconserver package on that service node, then set
+the B<console> column of B<servicenode> table to B<2>.
+
+To support diskless service node, B<makegocons> would help store the console state in the B<consoleenabled> column of B<nodehm>
```
Comment (gurevichmark) [inline xCAT-client/pods/man8/makegocons.8.pod]: It is a little confusing what "would help store" means here. Does the user need to do something to help "makegocons" ?

File: xCAT-client/pods/man8/makegocons.8.pod [hunk 5]
```diff
@@ -28,13 +26,17 @@ In the case of a hierarchical cluster (i.e. one with service nodes) B<makegocons
 which nodes will have their consoles accessed from the management node and which from a service node
 (based on the nodehm.conserver attribute).
 
+To start goconserver on the specified service node, please setup goconserver package on that service node, then set
```
Comment (gurevichmark) [inline xCAT-client/pods/man8/makegocons.8.pod]: Better not to use word "please"

File: xCAT-client/pods/man8/makegocons.8.pod [hunk 6]
```diff
@@ -4,9 +4,7 @@ B<makegocons> - Register or unregister the node in the goconserver service
 
 =head1 SYNOPSIS
 
-B<makegocons> [B<-V|--verbose>] [B<-d|--delete>] [I<noderange>]
-
-B<makeconservercf> [B<-h|--help|-v|--version>]
+B<makegocons> [B<-V|--verbose>] [B<-d|--delete>] [[B<-q|--query>]] [I<noderange>]
```
Comment (gurevichmark) [inline xCAT-client/pods/man8/makegocons.8.pod]: Seems like a double set of `[[` and `]]` around "query"

File: xCAT-client/pods/man8/makegocons.8.pod (hunk 2 above)
Comment (chenglch) [reply xCAT-client/pods/man8/makegocons.8.pod]: I don't know how to describe this kind of nodes accurately. I plan to use `console` nodes instead.
Only with `cons` attribute or `serialport` attribute defined, the nodes will be looked as a console nodes. This is default behavior from `makeconservercf`.

File: xCAT-client/pods/man8/makegocons.8.pod (hunk 4 above)
Comment (chenglch) [reply xCAT-client/pods/man8/makegocons.8.pod]: No, this is done by `makegocons` command automatically. I just want to mention that a new column `consoleenabled` has been added.

**Comment by chenglch** [comment]
@gurevichmark , I have modified this pull request based on your comments, could you help review again?

**Pull request merged**